tests/tracer/test_single_span_sampling_rules.py

```diff
@@ -119,6 +119,169 @@ def test_single_span_rules_do_not_tag_if_tracer_samples():
+    file.write('[{"sample_rate":0.5,"service":"xyz","name":"abc","max_per_second":100}]')
+
+    with override_env(dict(DD_SPAN_SAMPLING_RULES_FILE=str(file))):
+        sampling_rules = get_span_sampling_rules()
+        assert sampling_rules[0]._sample_rate == 0.5
+        assert sampling_rules[0]._service_matcher.pattern == "xyz"
+        assert sampling_rules[0]._name_matcher.pattern == "abc"
+        assert sampling_rules[0]._max_per_second == 100
+        assert len(sampling_rules) == 1
+
+
+def test_sampling_rule_init_config_multiple_sampling_rules_via_file(tmpdir):
+    file = tmpdir.mkdir("data").join("rules.json")
+    file.write(
+        '[{"service":"xy?","name":"a*c"}, \
+            {"sample_rate":0.5,"service":"my-service","name":"my-name", "max_per_second":20}]'
+    )
+
+    with override_env(dict(DD_SPAN_SAMPLING_RULES_FILE=str(file))):
+        sampling_rules = get_span_sampling_rules()
+        assert sampling_rules[0]._sample_rate == 1.0
+        assert sampling_rules[0]._service_matcher.pattern == "xy?"
+        assert sampling_rules[0]._name_matcher.pattern == "a*c"
+        assert sampling_rules[0]._max_per_second == -1
+
+        assert sampling_rules[1]._sample_rate == 0.5
+        assert sampling_rules[1]._service_matcher.pattern == "my-service"
+        assert sampling_rules[1]._name_matcher.pattern == "my-name"
+        assert sampling_rules[1]._max_per_second == 20
+        assert len(sampling_rules) == 2
+
+
+def test_sampling_rule_init_config_only_service_via_file(tmpdir):
+    file = tmpdir.mkdir("data").join("rules.json")
+    file.write('[{"service":"xyz"}]')
+    with override_env(dict(DD_SPAN_SAMPLING_RULES_FILE=str(file))):
+        sampling_rules = get_span_sampling_rules()
+        assert sampling_rules[0]._sample_rate == 1.0
+        assert sampling_rules[0]._service_matcher.pattern == "xyz"
+        assert sampling_rules[0]._max_per_second == -1
+        assert len(sampling_rules) == 1
+
+
+def test_sampling_rule_init_only_name_via_file(tmpdir):
+    file = tmpdir.mkdir("data").join("rules.json")
+    file.write('[{"name":"xyz"}]')
+    with override_env(dict(DD_SPAN_SAMPLING_RULES_FILE=str(file))):
+        sampling_rules = get_span_sampling_rules()
+        assert sampling_rules[0]._sample_rate == 1.0
+        assert sampling_rules[0]._name_matcher.pattern == "xyz"
+        assert sampling_rules[0]._max_per_second == -1
+        assert len(sampling_rules) == 1
+
+
+def test_exception_thrown_no_name_or_service_via_file(tmpdir):
+    file = tmpdir.mkdir("data").join("rules.json")
+    file.write('[{"sample_rate":1.0}]')
+    with override_env(dict(DD_SPAN_SAMPLING_RULES_FILE=str(file))):
+        with pytest.raises(ValueError):
+            sampling_rules = get_span_sampling_rules()
+            assert sampling_rules is None
+
+
+def test_exception_thrown_service_unsupported_char_via_file(tmpdir):
+    file = tmpdir.mkdir("data").join("rules.json")
+    file.write('[{"sample_rate":1.0}]')
```

Comment:
is this the correct rules for "unsupported_char" ?

Reply:
thanks for the catch!